macOS build failure due to ARFLAGS

```
Making install in miniz
make[3]: Nothing to be done for `install-exec-am'.
make[3]: Nothing to be done for `install-data-am'.
Making install in jenkins
  CC       lookup3.o
  AR       libjenkins.a
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ar: illegal option -- j
usage:  ar -d [-TLsv] archive file ...
	ar -m [-TLsv] archive file ...
	ar -m [-abiTLsv] position archive file ...
	ar -p [-TLsv] archive [file ...]
	ar -q [-cTLsv] archive file ...
	ar -r [-cuTLsv] archive file ...
	ar -r [-abciuTLsv] position archive file ...
	ar -t [-TLsv] archive [file ...]
	ar -x [-ouTLsv] archive [file ...]
make[2]: *** [libjenkins.a] Error 1
make[1]: *** [install-recursive] Error 1
make: *** [install-recursive] Error 1
```

This is what happens if I remove
```
ENV["ARFLAGS"] = "cru"
```
from the Homebrew formula.

The comment in the formula says:
```
     # Remove as soon as possible
     # https://github.com/pioneerspacesim/pioneer/issues/3839
```
But https://github.com/pioneerspacesim/pioneer/issues/3839 is apparently closed (via https://github.com/pioneerspacesim/pioneer/pull/3987) and yet the problem remains.

https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/6489/version=sierra/console

--- a/configure.ac
+++ b/configure.ac
@@ -83,7 +83,8 @@ if test -z "$ARFLAGS"
 then
   case "${host_os}" in
   darwin*)
-  # do nothing, darwin does not support ar flag D
+  # darwin uses it's own version of ar with different flags to what GNU uses
+  ARFLAGS="r"
   ;;
   *)
   ARFLAGS="rcD"

